src/Operation/Division.php

```diff
@@ -0,0 +1,169 @@
+     * @throws DivisionByZeroException
+     */
+    public function compute(DecimalNumber $a, DecimalNumber $b, $precision = self::DEFAULT_PRECISION)
+    {
+        if (function_exists('bcdiv')) {
+            return $this->computeUsingBcMath($a, $b, $precision);
+        }
+
+        return $this->computeWithoutBcMath($a, $b, $precision);
+    }
+
+    /**
+     * Performs the division using BC Math
+     *
+     * @param DecimalNumber $a Dividend
+     * @param DecimalNumber $b Divisor
+     * @param int $precision Maximum decimal precision
+     *
+     * @return DecimalNumber Result of the division
+     * @throws DivisionByZeroException
+     */
+    public function computeUsingBcMath(DecimalNumber $a, DecimalNumber $b, $precision = self::DEFAULT_PRECISION)
+    {
+        if ((string) $b === '0') {
+            throw new DivisionByZeroException();
+        }
+
+        return new DecimalNumber((string) bcdiv($a, $b, $precision));
+    }
+
+    /**
+     * Performs the division without BC Math
+     *
+     * @param DecimalNumber $a Dividend
+     * @param DecimalNumber $b Divisor
+     * @param int $precision Maximum decimal precision
+     *
+     * @return DecimalNumber Result of the division
+     * @throws DivisionByZeroException
+     */
+    public function computeWithoutBcMath(DecimalNumber $a, DecimalNumber $b, $precision = self::DEFAULT_PRECISION)
+    {
+        $bString = (string) $b;
+        if ('0' === $bString) {
+            throw new DivisionByZeroException();
+        }
+
+        $aString = (string) $a;
+
+        // 0 as dividend always yields 0
+        if ('0' === $aString) {
+            return $a;
+        }
+
+        // 1 as divisor always yields the dividend
+        if ('1' === $bString) {
+            return $a;
+        }
+
+        // -1 as divisor always yields the the inverted dividend
+        if ('-1' === $bString) {
+            return $a->invert();
+        }
+
+        // if dividend and divisor are equal, the result is always 1
+        if ($a->equals($b)) {
+            return new DecimalNumber('1');
+        }
+
+        $aPrecision = $a->getPrecision();
+        $bPrecision = $b->getPrecision();
+
+        $maxPrecision = max($aPrecision, $bPrecision);
+
+        if ($maxPrecision > 0) {
+            // multiply $a and $b by 10^(number of decimal places of b)
+            $a = $a->toMagnitude($maxPrecision);
+            $b = $b->toMagnitude($maxPrecision);
+        }
+
+        $result = $this->integerDivision($a, $b, max($precision, $aPrecision));
+
+        return $result;
+    }
+
+    /**
+     * Computes the division between to integer DecimalNumbers
+     *
+     * @param DecimalNumber $a Dividend
+     * @param DecimalNumber $b Divisor
+     * @param int $precision Maximum number of decimals to try
+     *
+     * @return DecimalNumber
+     */
+    private function integerDivision(DecimalNumber $a, DecimalNumber $b, $precision)
+    {
+        $dividend = $a->getCoefficient();
```

Comment:
No need to because I make both the divisor and dividend integers here: https://github.com/PrestaShop/decimal/blob/7fa52f14ccf1eb09cc761b5945b9df3571efb86b/src/Operation/Division.php#L103-L107

I'll add a comment about that for clarification.